### PR TITLE
[ESIMD] Add tests for DPAS with tfloat32 arguments

### PR DESCRIPTION
**Pull request by v-klochkov**
This patch also enables those test-cases that previously caused errors being started on simulator.

Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+//==---------------- dpas_tf32.cpp  - DPC++ ESIMD on-device test ----------===//
```
Comment (kbobrovs) [inline SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp:1]: why create new test in deprecated dir?

File: SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc || esimd_emulator
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -DESIMD_XE_HPC %s -o %t.out
```
Comment (kbobrovs) [inline SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp]: ESIMD_XE_HPC should not be needed

File: SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: esimd_emulator
```
Comment (kbobrovs) [inline SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp:12]: @dongkyunahn-intel, FYI

File: SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp (hunk 1 above)
Comment (v-klochkov) [reply SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp:1]: The new test (dpas_tf32.cpp) follows the same pattern that is used for other _new_ tests like dpas_bf16.cpp, but that name dpas_tf32.cpp was already taken by old test, that I decided to keep and move to deprecated/dpas_tf32.cpp.
Other old tests are also moved to 'deprecated'

File: SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp (hunk 2 above)
Comment (v-klochkov) [reply SYCL/ESIMD/dpas/deprecated/dpas_tf32.cpp]: Right, Thank you! I removed usage of those macro from tests here: 4240b1334a875eec376dbcd97cdbc3f47a89dfac
Also, adjusted paths to esimd_test_utils.hpp in the dpas/deprecated/* tests.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**